StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/LinkPaymentController.swift

```diff
@@ -201,20 +230,153 @@ import UIKit
                     case .failure(let error):
-                        self.paymentMethodId = nil
-                        self.payWithLinkContinuation?.resume(throwing: error)
+                        self.continueWithFailure(error)
                     }
                     self.payWithLinkContinuation = nil
                 }
         }
     }
+    
+    private func presentWithSDK(
+        from presentingViewController: UIViewController,
+        completionHandler: @escaping (FinancialConnectionsSDKResult?, LinkAccountSession?, NSError?) -> Void
+    ) {
+        let bankAccountCollector = STPBankAccountCollector()
+        
+        let additionalParameters: [String: Any] = [
+            "product": "instant_debits",
+        ]
+        
+        let params = STPCollectBankAccountParams.collectInstantDebitsParams(
+            email: configuration.defaultBillingDetails.email
+        )
+        
+        let elementsSessionContext = makeElementsSessionContext()
+        
+        switch mode {
+        case .paymentIntentClientSecret(let string):
+            bankAccountCollector.collectBankAccountForPayment(
+                clientSecret: string,
+                returnURL: configuration.returnURL,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
+                onEvent: nil,
+                params: params,
+                from: presentingViewController,
+                financialConnectionsCompletion: completionHandler
+            )
+        case .setupIntentClientSecret(let string):
+            bankAccountCollector.collectBankAccountForSetup(
+                clientSecret: string,
+                returnURL: configuration.returnURL,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
+                onEvent: nil,
+                params: params,
+                from: presentingViewController,
+                financialConnectionsCompletion: completionHandler
+            )
+        case .deferredIntent(let intentConfiguration):
+            let amount: Int?
+            let currency: String?
+            switch intentConfiguration.mode {
+            case let .payment(amount: _amount, currency: _currency, _, _):
+                amount = _amount
+                currency = _currency
+            case let .setup(currency: _currency, _):
+                amount = nil
+                currency = _currency
+            }
+            bankAccountCollector.collectBankAccountForDeferredIntent(
+                sessionId: "ios_instant_debits_only_\(UUID().uuidString)",
+                returnURL: configuration.returnURL,
+                onEvent: nil,
+                amount: amount,
+                currency: currency,
+                onBehalfOf: nil,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
```

Comment:
Do we want to pass the `amount` / `currency` to the `elementsSessionContext` since we have it? It gets passed through to the link sign up API call!

Reply:
Done!